src/clj/game/cards/identities.clj

```diff
@@ -1651,3 +1669,21 @@
              :effect (effect (move :runner (last (:discard runner)) :deck)
                              (shuffle! :runner :deck)
                              (trigger-event :searched-stack nil))}]})
+
+(defcard "Zahya Sadeghi: Versatile Smuggler"
+  {:events [{:event :run-ends
+             :req (req (and (or (= :hq (target-server context))
+                                (= :rd (target-server context)))
+                            (pos? (total-cards-accessed context))))
+             :effect (effect (continue-ability
+                               (let [cards-accessed (total-cards-accessed context)]
```

Comment:
This is exposed in the optional, because we pass in `targets` to both the prompt and the `yes-ability`. So the `continue-ability` nesting can be removed so it's purely `{:event :run-ends :optional {:req ...`